testing/trino-tests/src/test/java/io/trino/execution/TestEventListenerWithSplits.java

```diff
@@ -122,7 +123,10 @@ public void testSplitsForNormalQuery()
         QueryCompletedEvent queryCompletedEvent = queryEvents.getQueryCompletedEvent();
         assertThat(queryCompletedEvent.getContext().getResourceGroupId().isPresent()).isTrue();
         assertThat(queryCompletedEvent.getContext().getResourceGroupId().get()).isEqualTo(createResourceGroupId("global", "user-user"));
-        assertThat(queryCompletedEvent.getIoMetadata().getOutput()).isEqualTo(Optional.empty());
+        Optional<QueryOutputMetadata> output = queryCompletedEvent.getIoMetadata().getOutput();
```

Comment:
Instead of capturing this event for `SELECT` in  `QueryOutputMetadata` ? Can we add additional field in `QueryIOMetadata` where there could be a `Optional<List<OutputColumnMetadata>>` which could be set only for `SELECT` statements in this way the logic could be simplified. 

@martint - Is `QueryIOMetadata` a right place to add them or is there a better place we could store them ?